docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,138 @@
+        * It takes as input a [KeptnMetric](../../reference/crd-reference/metric.md)
+          and [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md)
+      * `EvaluateQueryForStep`(Fetches metric values with step interval from the provider)
+        * This function fetches metric values with a specified step interval from the placeholder provider.
+          It takes into account the metric query and the step interval provided, executes the query,
+          and returns the metric values along with any additional data if required.
+        * It takes as input a [KeptnMetric](../../reference/crd-reference/metric.md)
+          and [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md)
+      * `FetchAnalysisValue`(Fetches analysis values from the provider) functions.
+        * This function fetches analysis values based on the provided query and time range from the
+          provider.
+          It evaluates the query within the specified time range and returns the analysis
+          values along with any additional data if required.
+        * It takes as input an [Analysis](../../reference/crd-reference/analysis.md),
+          resource that contains a `query` and a
+          [KeptnMetricsProvider](../../reference/crd-reference/metricsprovider.md) resource.
+
+     You can follow other existing implementations,
+     such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+     as an example.
+
+     Each of the three functions expects a string containing a float value in it.
+     But for example purposes we returned some of the data accessible in the function.
+
+     Below is an example of a placeholder provider implementation.
+
+      ```go
+        {% include "./assets/example-code/placeholder-code-example.go" %}
+      ```
+
+      > **Note** Refer to the documentation of the
+      > [KeptnMetric](../../reference/crd-reference/metric.md)
+      > and
+      > [Analysis](../../reference/crd-reference/analysis.md)
+      > resources
+      > to understand what data should be retrieved from the methods inputs to compute accurate results.
+
+4. **Instantiate the Provider** in the `providers.NewProvider` function
+   in the `metrics-operator/controllers/common/providers/provider.go` file.
+   add a case for the `KeptnPlaceholderProviderType`.
+   Instantiate the placeholder provider struct and return it.
+
+    ```go
+    // Inside the providers package
+
+    // NewProvider function
+    func NewProvider(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+        switch strings.ToLower(providerType) {
+        case KeptnPlaceholderProviderType:
+            return &placeholder.KeptnPlaceholderProvider{
+                Log:        log,
+                HttpClient: http.Client{},
+            }, nil
+        // Other cases...
+        }
+    }
```

Comment:
Please move this code snippet to a separate file and link it via include (same as done [here](https://github.com/keptn/lifecycle-toolkit/pull/2939/files#diff-d41d4c55a5a07b73a44c5db5a01e826b0a59ee530fcd12d7fa646d7e2b413759R75))

Reply:
Alright